spring-ai-core/src/main/java/org/springframework/ai/model/function/FunctionCallbackContext.java

```diff
@@ -150,6 +179,15 @@ public enum SchemaType {
 
 	private static class KotlinDelegate {
 
+		public static boolean isKotlinSupplier(Class<?> clazz) {
+			return Function0.class.isAssignableFrom(clazz);
+		}
+
+		@SuppressWarnings("unchecked")
+		public static Supplier<?> wrapKotlinSupplier(Object function) {
+			return () -> ((Function0<Object>) function).invoke();
+		}
```

Comment:
Likely we may have to add support for the Kotlin's equivalent of java.util.Consumer() in a follow up PR.